service/history/workflowExecutionContext.go

```diff
@@ -576,6 +610,14 @@ func (c *workflowExecutionContextImpl) updateWorkflowExecutionWithNew(
 		if err != nil {
 			return err
 		}
+		if updateMode == persistence.UpdateWorkflowModeBypassCurrent {
```

Comment:
this logic should be called right before `updateWorkflowExecutionWithRetry`

Reply:
done